### PR TITLE
stale-while-revalidate

### PR DESCRIPTION
**Pull request by gburgett**
* Rewrite stale while revalidate

### CODE REVIEW

**Comment by gburgett** [comment]
Tested with some example queries from the logs.
```
# With cache control:
curl -I -XGET 'http://localhost:4000/?query=query%20getChildContent(%24itemId%3A%20ID!%2C%20%24showLabel%3A%20Boolean!)%20%7B%0A%20%20node(id%3A%20%24itemId)%20%7B%0A%20%20%20%20...%20on%20ContentItem%20%7B%0A%20%20%20%20%20%20id%0A%20%20%20%20%20%20childContentItemsConnection%20%7B%0A%20%20%20%20%20%20%20%20edges%20%7B%0A%20%20%20%20%20%20%20%20%20%20node%20%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20id%0A%20%20%20%20%20%20%20%20%20%20%20%20coverImage%20%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20name%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20sources%20%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20uri%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20__typename%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20__typename%0A%20%20%20%20%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%20%20%20%20%20%20%20%20title%0A%20%20%20%20%20%20%20%20%20%20%20%20summary%0A%20%20%20%20%20%20%20%20%20%20%20%20...%20on%20Event%20%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20label%20%40include(if%3A%20%24showLabel)%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20__typename%0A%20%20%20%20%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%20%20%20%20%20%20%20%20__typename%0A%20%20%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%20%20%20%20%20%20__typename%0A%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%20%20%20%20__typename%0A%20%20%20%20%20%20%7D%0A%20%20%20%20%20%20__typename%0A%20%20%20%20%7D%0A%20%20%20%20__typename%0A%20%20%7D%0A%7D%0A&operationName=getChildContent&variables=%7B%22itemId%22%3A%22Announcement%3A12a556a5f37034d23ad0b55fe22ed0f989e7d4774fa951a890c4a41965a8387e%22%2C%22showLabel%22%3Atrue%7D'
HTTP/1.1 200 OK
X-Powered-By: Express
Access-Control-Allow-Origin: *
Content-Type: application/json; charset=utf-8
cache-control: max-age=3600, public, stale-while-revalidate=30, stale-if-error=86400

# Where cache control is zero:
curl -I -XGET 'http://localhost:4000/?query=query%20currentUserId%20%7B%0A%20%20currentUser%20%7B%0A%20%20%20%20id%0A%20%20%20%20__typename%0A%20%20%7D%0A%7D%0A&operationName=currentUserId&variables=%7B%7D'
HTTP/1.1 200 OK
X-Powered-By: Express
Access-Control-Allow-Origin: *
Content-Type: application/json; charset=utf-8
Content-Length: 1676
ETag: W/"68c-tCJIQZ5YcxFsKT96koF3tY3ISIQ"
Date: Thu, 22 Apr 2021 16:46:47 GMT
Connection: keep-alive
```

File: apollos-church-api/src/server.js [hunk 1]
```diff
@@ -36,7 +36,7 @@ const isDev =
 
 const extensions = isDev ? [() => new RockLoggingExtension()] : [];
 
-const cacheOptions = isDev
+const cacheOptions = !isDev
```
Comment (conrad-vanl) [inline apollos-church-api/src/server.js:39]: why the bang here? This will clear out cache options in production

Comment (gburgett) [reply apollos-church-api/src/server.js:39]: whoops - forgot to undo this.  Thanks for the code review!

File: apollos-church-api/src/server.js [hunk 2]
```diff
@@ -90,6 +90,22 @@ app.get('/version', cors(), (req, res) => {
   }
 });
 
+app.use('/', (req, res, next) => {
```
Comment (conrad-vanl) [inline apollos-church-api/src/server.js:93]: do you need the `/` - does this work without providing a path? I'm trying to remember how the matching works.... both requests to `/` and `/graphql` are valid and I'm not sure if the mobile client uses one or the other

Comment (gburgett) [reply apollos-church-api/src/server.js:93]: Ah ok, I don't think I need that path.  Didn't want to over-specify!

**Review by conrad-vanl — APPROVED** [review]

File: apollos-church-api/src/server.js (hunk 2 above)
Comment (gburgett) [reply apollos-church-api/src/server.js]: Cool - just tested it works at both root and `/graphql`

**Pull request merged**